cranelift-codegen/src/isa/x86/unwind.rs

```diff
@@ -0,0 +1,525 @@
+//! Unwind information for x64 Windows.
+use super::registers::RU;
+use crate::ir::{Function, InstructionData, Opcode};
+use crate::isa::{CallConv, RegUnit, TargetIsa};
+use alloc::vec::Vec;
+
+// Maximum (inclusive) size of a "small" stack allocation
+const SMALL_ALLOC_MAX_SIZE: u32 = 128;
+// Maximum (inclusive) size of a "large" stack allocation that can represented in 16-bits
+const LARGE_ALLOC_16BIT_MAX_SIZE: u32 = 524280;
+
+struct MemoryWriter {
+    ptr: *mut u8,
+    offset: usize,
+}
+
+impl MemoryWriter {
+    pub fn write_u8(&mut self, v: u8) {
+        self.write(v);
+    }
+
+    pub fn write_u16(&mut self, v: u16) {
+        self.write(v);
+    }
+
+    pub fn write_u32(&mut self, v: u32) {
+        self.write(v);
+    }
+
+    fn write<T>(&mut self, x: T) {
+        unsafe {
+            #[cfg_attr(feature = "cargo-clippy", allow(clippy::cast_ptr_alignment))]
```

Comment:
This should probably always write little endian so that the host endianness doesn't affect it.